### PR TITLE
Refactor build to use PSPackageProject

### PR DESCRIPTION
**Pull request by PaulHigin**
This is the first build refactoring that arranges PowerShellGet to use PSPackageProject module for build/test/compliance/signing/publishing.

- Files have been moved to their needed locations

- The build script has been refactored to conform to PSPackageProject 

- CI yml files have been added but not fully changed for PowerShellGet and not yet tested

- Cmldet help files have been moved to location that conforms with PSPackageProject

### CODE REVIEW

**Comment by alerickson** [comment]
There's already a build file in [src](https://github.com/PowerShell/PowerShellGet/blob/development/src/build.ps1), can we merge the two files (either getting rid of that build file or the one in this repo)?

**Comment by alerickson** [comment]
@sdwheeler had opened a [PR ](https://github.com/PowerShell/PowerShellGet/pull/214) adding all the documentation files into 'docs'.  Do these files explicitly need to be in a folder called 'docs' or is moving them to 'help' okay?

**Comment by alerickson** [comment]
> There's already a build file in [src](https://github.com/PowerShell/PowerShellGet/blob/development/src/build.ps1), can we merge the two files (either getting rid of that build file or the one in this repo)?

Nevermind, just saw that you removed the other build script

File: .ci/ci.yml [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+trigger:
+  # Batch merge builds together while a merge build is running
+  batch: true
+  branches:
+    include:
+    - master
+pr:
+  branches:
+    include:
+    - master
```
Comment (alerickson) [inline .ci/ci.yml:11]: Is this setting up CI explicitly for the master branch?  If so can we change this to development?

Comment (PaulHigin) [reply .ci/ci.yml:11]: CI is not running yet, but when it does it will run on any selected branch.  I am still working to get a project set up.

**Comment by PaulHigin** [comment]
I updated the tests to work with Pester 4.x.  I think we should standardize on this version as the rest of PowerShell/Module tests already do.  There are a number of breaking changes in 5.x that make CI difficult.

Also I temporarily disabled the older tests since they are not running successfully and also change machine state.  I think they need to be reviewed.

**Review by alerickson — APPROVED** [review]

**Review by anamnavi — APPROVED** [review]

**Comment by anamnavi** [comment]
messed Paul about the one question I had and got a better understanding. The rest of the PR looks good to me :)

File: README.md [hunk 2]
```diff
@@ -7,65 +7,95 @@
...
-Import-Module .\bin\Debug\netstandard2.0\publish\PowerShellGet.dll
+C:\> Import-Module C:\Repos\PowerShellGet\out\PowerShellGet
 
-# if running Windows PowerShell
-Import-Module .\bin\Debug\netstandard2.0\publish\PowerShellGet.dll
+# If running Windows PowerShell
+C:\> Import-Module C:\Repos\PowerShellGet\out\PowerShellGet\PowerShellGet.psd1
 ```
 
+**Note**
```
Comment (alerickson) [inline README.md:98]: Thanks for adding this!

**Pull request merged**